asynciterator.ts

```diff
@@ -81,6 +81,46 @@ export class AsyncIterator<T> extends EventEmitter {
   private _readable = false;
   protected _properties?: { [name: string]: any };
   protected _propertyCallbacks?: { [name: string]: [(value: any) => void] };
+  private _readableFlowing: boolean | null = null;
```

Comment:
I wonder if it is a necessity to maintain this flag, but let's keep it for now and see if we can remove/simplify it after the unit tests pass.